### PR TITLE
Add data source for listing AWS integrations 

### PR DESCRIPTION
**Pull request by KNiepok**
## Description of the change

> This PR introduces a data source spacelift_aws_integrations that can return all of the users AWS integrations with schema similar to the singular integration.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (non-breaking change that adds documentation)

## Related issues

> Fix [#1]()

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development
- [ ] Examples for new resources and data sources have been added
- [ ] Default values have been documented in the description (e.g., "Dummy: (Boolean) Blah blah. Defaults to `false`.)
- [ ] If the action fails that checks the documentation: Run `go generate` to make sure the docs are up to date

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] Pull Request is no longer marked as "draft"
- [ ] Reviewers have been assigned
- [ ] Changes have been reviewed by at least one other engineer
- [ ] The target branch is `future` unless the change is going directly into production

### CODE REVIEW

**Comment by marcinwyszynski** [comment]
Task linked: [CU-2ry65hb Add data source to list all AWS/Azure integrations](https://app.clickup.com/t/2ry65hb)

File: spacelift/data_aws_integrations.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/validations"
+)
+
+func dataAWSIntegrations() *schema.Resource {
+	return &schema.Resource{
+		Description: "`spacelift_aws_integration` represents a list of all the AWS integrations in the Spacelift account",
```
Comment (TheMacies) [inline spacelift/data_aws_integrations.go:16]: in the Spacelift account -> in the Spacelift account visible to the API user. The reason is with spaces you might see just a subset.

File: spacelift/data_aws_integrations.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+		Schema: map[string]*schema.Schema{
+			"integrations": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"integration_id": {
+							Type:             schema.TypeString,
+							Description:      "Immutable ID of the integration.",
+							Optional:         true,
```
Comment (adamconnelly) [inline spacelift/data_aws_integrations.go:27]: Why are the `integration_id` and `name` optional? Shouldn't they just be computed - it doesn't look like we're using the values for querying.

**Review by adamconnelly — COMMENTED** [review]
Looks good, although I don't think the integration_id and name should be settable.

File: spacelift/data_aws_integrations_test.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+)
+
+func TestAWSIntegrationsData(t *testing.T) {
+	randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
+	resourceName := "spacelift_aws_integration.test"
+	datasourceName := "data.spacelift_aws_integrations.test"
+
+	testSteps(t, []resource.TestStep{{
+		Config: fmt.Sprintf(`
+		 resource "spacelift_aws_integration" "test" {
```
Comment (adamconnelly) [inline spacelift/data_aws_integrations_test.go:20]: I think it's worth creating multiple integrations in the test. That way we'll have confidence the code that maps the result to the TF objects isn't doing anything weird.

File: spacelift/data_aws_integrations.go (hunk 2 above)
Comment (KNiepok) [reply spacelift/data_aws_integrations.go]: Correct, I removed those

File: spacelift/data_aws_integrations_test.go (hunk 3 above)
Comment (KNiepok) [reply spacelift/data_aws_integrations_test.go]: I did an overhaul of the tests. Now the test checks all the fields and it also creates two resources (and checks both of them)

File: spacelift/data_aws_integrations.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+)
+
+func dataAWSIntegrations() *schema.Resource {
+	return &schema.Resource{
+		Description: "`spacelift_aws_integration` represents a list of all the AWS integrations  in the Spacelift account visible to the API user.",
```
[the remark this answers is not in the document]
Comment (TheMacies) [reply spacelift/data_aws_integrations.go:15]: also  there is a double space between `..integrations  in...`

**Review by TheMacies — APPROVED** [review]
LGTM, left just some minor comments about description

**Review by adamconnelly — APPROVED** [review]
LGTM - nice one!

**Pull request merged**